drools-wb-screens/drools-wb-test-scenario-editor/drools-wb-test-scenario-editor-client/src/main/java/org/drools/workbench/screens/testscenario/client/page/audit/AuditPageViewImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.testscenario.client.page.audit;
+
+import java.util.Arrays;
+import java.util.Set;
+
+import javax.enterprise.context.Dependent;
+import javax.inject.Inject;
+
+import elemental2.dom.HTMLDivElement;
+import elemental2.dom.HTMLElement;
+import org.drools.workbench.models.testscenarios.shared.ExecutionTrace;
+import org.drools.workbench.screens.testscenario.client.resources.i18n.TestScenarioConstants;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Templated
+@Dependent
+public class AuditPageViewImpl implements AuditPage.AuditPageView {
+
+    @DataField("root")
+    private HTMLDivElement root;
+
+    @DataField("audit-log")
+    private HTMLDivElement auditLogDiv;
+
+    @DataField("fired-rules")
+    private HTMLDivElement firedRulesDiv;
+
+    private AuditPage presenter;
+
+    private AuditTable firedRulesTable;
+
+    private AuditTable auditLogTable;
+
+    @Inject
+    public AuditPageViewImpl(final HTMLDivElement root,
+                             final HTMLDivElement auditLogDiv,
+                             final HTMLDivElement firedRulesDiv,
+                             final AuditTable firedRulesTable,
+                             final AuditTable auditLogTable) {
+        this.root = root;
+        this.auditLogDiv = auditLogDiv;
+        this.firedRulesDiv = firedRulesDiv;
+        this.firedRulesTable = firedRulesTable;
+        this.auditLogTable = auditLogTable;
+
+        this.firedRulesDiv.appendChild(firedRulesTable.getElement());
```

Comment:
I've yet to write a Elemental2 screen so am guessing here; but can you not add the Rules Table to the Audit Page using the HTML? If your ```@DataField``` annotations were against the two ```AuditTable``` fields and you removed the "containing" ```HTMLDivElement```s? What you've done here is in essence what ```errai-ui``` does.

Reply:
```AuditTable``` implements ```IsElement``` so should be "injectable" into the HTML.

Reply:
@mbarkley Is my understanding correct? 

We should be able to have ```errai-ui``` _inject_ **any** ```IsElement``` into the HTML template?

Reply:
@manstis Yes, you should be able to put the `AuditTable` as a `@DataField` and have the corresponding `data-field` element be a child of the `data-field="fired-rules"` element in your HTML template. This should work if the `AuditTable` is any kind of UI component (Errai UI `@Templated` bean, GWT Widget, Elemental2 IsElement, Errai DOM Wrapper IsElement).

In general the only reason you *might* ever need to manually construct parts of the DOM with Errai UI is for dynamic UI components (i.e. entries in a list or table, populated from somewhere else).

Reply:
Great, will apply.